pkg/controller/main-controller.go

```diff
@@ -304,6 +304,9 @@ func NewController(
 			}
 			controller.enqueueTenant(newObj)
 		},
+		// Enqueue tenant to perform some delete handling actions
+		// during reconciliation
+		DeleteFunc: controller.enqueueTenant,
```

Comment:
I don't see any actual deletes being added. It resources need to be removed when the `Tenant` is removed, then they should use the `Tenant` as the owner of the resource, so deletion is automatic.